java/src/main/java/org/rocksdb/RocksDB.java

```diff
@@ -61,35 +63,39 @@ public static synchronized void loadLibrary() {
    *     of a library.
    */
   public static synchronized void loadLibrary(final List<String> paths) {
-    for (CompressionType compressionType : CompressionType.values()) {
-      if (compressionType.equals(CompressionType.NO_COMPRESSION)) {
-        continue;
+    if(!libraryLoaded) {
```

Comment:
Two functions are accessing and modifying libraryLoaded.  Does `synchronized` use same lock on all functions or each function has its own lock?  If its the later case then it might be a good idea to have these two functions share the same code-path.